js/client-chat.js

```diff
@@ -567,11 +567,20 @@
 					for (var i=0, len=targets.length; i<len; i++) {
 						targets[i] = targets[i].trim();
 					}
-					switch (targets[0]) {
+					switch (targets.shift()) {
 					case 'add':
-						for (var i=1, len=targets.length; i<len; i++) {
-							highlights.push(targets[i].trim());
-						}
```

Comment:
1. internally, `concat` loops through `targets`, anyway; it's only slightly faster than actually writing out the loop.

2. `targets[i].trim()` is applied, which your code doesn't do.